js/adapt-close.js

```diff
@@ -46,7 +46,13 @@ define([ "core/js/adapt" ], function(Adapt) {
 			var config = Adapt.course.get("_close");
 			config.browserPromptIfIncomplete = config.browserPromptIfComplete = false;
 
-			top.window.close();
+			if(config._button._closeViaLMSFinish) {
+				$(window)

```

Comment:
could you change lines 50 - 52 over to this:
```js
var scorm = require('extensions/adapt-contrib-spoor/js/scorm/wrapper');
if (scorm) scorm.getInstance().finish();
```
?

It should have the same effect, only it will be more robust.